clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -80,4 +81,18 @@ class ApiClient(apiUrl: String, accessKey: String, secretKey: String) {
         throw new RuntimeException(s"expected string commit_id in ${resp.body}")
     }
   }
+
+  def uploadIfAbsent(repoName: String, branch: String, path: String, stream: InputStream, byteSize: Long): Unit = {
+    val uploadObjectURI = URI.create("%s/repositories/%s/branches/%s/objects?path=%s".format(apiUrl, repoName, branch, path)).normalize()
```

Comment:
I don't think `URI.create` does any escaping.
Since we don't reject unknown query strings, if the path contains a `&`, we'll silently upload to an incorrect path.